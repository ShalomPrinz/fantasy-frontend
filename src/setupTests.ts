// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { ReactElement } from 'react';

import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

export type TestRendererJSON = null | ReactTestRendererJSON | ReactTestRendererJSON[];
export const getJSON = (Component: ReactElement) => renderer.create(Component).toJSON()
