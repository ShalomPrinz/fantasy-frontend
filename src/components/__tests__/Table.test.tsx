import { Table } from '../';
import { render, TestComponent } from '../../setupTests'

let root: TestComponent = null;

beforeEach(() => {
    root = null
})

describe('Table', () => {
    it('should render a scrollable table', () => {
        const data = [{ id: 1, props: { label: 'Label' } }]
        const columns = [{ id: 0, path: 'label'}]

        root = render(<Table data={data} columns={columns} scrollable />);

        // @ts-ignore root should never be null
        expect(root.toJSON()).toMatchSnapshot();
    })
    
    it('should render a styled table', () => {
        const data = [{ id: 1, props: { label: 'Label' } }]
        const columns = [{ id: 0, path: 'label'}]

        root = render(<Table data={data} columns={columns} className="bg-default" />)

        // @ts-ignore root should never be null
        expect(root.toJSON()).toMatchSnapshot();
    })

    it('should render a column by content callback', () => {
        const data = [{ id: 1, props: { label: 'Label' } }]
        const content = ({ label }: { [key: string]: string }) => (<h1>{label}</h1>)
        const columns = [{ id: 0, path: 'label', content: content}]

        root = render(<Table data={data} columns={columns} className="bg-default" />)

        // @ts-ignore root should never be null
        expect(root.toJSON()).toMatchSnapshot();
    })
})
