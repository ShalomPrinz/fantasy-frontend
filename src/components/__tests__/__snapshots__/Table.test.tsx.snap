// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render a scrollable table 1`] = `
<div
  className="table-responsive mt-0 border-bottom scrollable"
>
  <table
    className="table rounded"
  >
    <tbody>
      <tr>
        <td
          className="align-middle"
        >
          Label
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table should render a styled table 1`] = `
<div
  className="table-responsive mt-0"
>
  <table
    className="table rounded bg-default"
  >
    <tbody>
      <tr>
        <td
          className="align-middle"
        >
          Label
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
